packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -548,9 +548,9 @@ export default class NewVm extends BaseComponent {
     const defaultNetworkIds = this._getDefaultNetworkIds(template)
     forEach(template.VIFs, vifId => {
       const vif = getObject(storeState, vifId, resourceSet)
-      VIFs.push({
-        network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0],
-      })
+      // insertion of the vif at the index that corresponds to the order chosen by the user when creating the VM
+      // this order is stored in the key named "device" of the VIF object
+      VIFs.splice(vif.device, 0, {network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0]})
```

Comment:
According to the XAPI [documentation](https://xapi-project.github.io/xen-api/classes/vif.html), `VIF.device` is a string. Isn't it better to convert it to a number to use as an index?
```suggestion
      VIFs.splice(Number(vif.device), 0, {network: pool || isInResourceSet(vif.$network) ? vif.$network : defaultNetworkIds[0]})
```

Reply:
And handle if `NaN` after conversion.